### PR TITLE
cn, work, governance: change TxPool into interface

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- To mock TxPool in `cn.CN` and `sc.ServiceChain`, `work.TxPool` interface has been introduced.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: node/cn/protocol.go [hunk 1]
```diff
@@ -117,23 +115,6 @@ var errorToString = map[int]string{
...
-	// The slice should be modifiable by the caller.
-	Pending() (map[common.Address]types.Transactions, error)
-
-	CachedPendingTxsByCount(count int) types.Transactions
-
-	// SubscribeNewTxsEvent should return an event subscription of
-	// NewTxsEvent and send events to the given channel.
-	SubscribeNewTxsEvent(chan<- blockchain.NewTxsEvent) event.Subscription
-}
-
```
Comment (KimKyungup) [inline node/cn/protocol.go]: It seems that this `TxPool` interface is integrated with `work.TxPool`. 
Why do you merge them? Merged interface is better than separated interface for several packages?

Comment (ehnuje) [reply node/cn/protocol.go]: @KimKyungup Thanks for the comment. As `cn.CN` and `sc.ServiceChain` implement `work.Backend` interface, they need to use the same type for `work.TxPool`. This is the reason why `work.TxPool` interface has been introduced.

Comment (KimKyungup) [reply node/cn/protocol.go]: I misunderstood. :) Thanks.

File: node/cn/handler_test.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ import (
 	"github.com/klaytn/klaytn/node"
 	"github.com/klaytn/klaytn/node/cn/mocks"
 	"github.com/klaytn/klaytn/params"
+	mocks3 "github.com/klaytn/klaytn/work/mocks"
```
Comment (jeonghonah) [inline node/cn/handler_test.go]: How about this?
```suggestion
	workMocks "github.com/klaytn/klaytn/work/mocks"
```

File: work/work.go [hunk 3]
```diff
@@ -34,16 +34,43 @@ import (
...
 	"github.com/klaytn/klaytn/params"
 	"github.com/klaytn/klaytn/storage/database"
+	"math/big"
 	"sync/atomic"
 )
 
 var logger = log.NewModuleLogger(log.Work)
 
+//go:generate mockgen -destination=work/mocks/txpool_mock.go -package=mocks github.com/klaytn/klaytn/work TxPool
+// TxPool is an interface of blockchain.TxPool used by ProtocolManager.
```
Comment (jeonghonah) [inline work/work.go]:
```suggestion
// TxPool is an interface of blockchain.TxPool used by ProtocolManager and backends.
```

File: work/work.go [hunk 4]
```diff
@@ -34,16 +34,43 @@ import (
...
+type TxPool interface {
```
Comment (jeonghonah) [inline work/work.go]: Minor comment:
`package work` is for block creation and mining. Is it okay to add `TxPool` in this package?

Comment (ehnuje) [reply work/work.go]: Could you please explain in detail? What can be different if `TxPool` interface is declared in another package?

File: node/cn/handler_test.go (hunk 2 above)
Comment (ehnuje) [reply node/cn/handler_test.go]: Updated!

File: work/work.go (hunk 3 above)
Comment (ehnuje) [reply work/work.go]: Updated!

File: work/work.go (hunk 4 above)
Comment (jeonghonah) [reply work/work.go]: Assuming that you are looking for where the TxPool interface is defined, I will find the blockchain package first and then search other packages. In this case, `package work` is unlikely to be on a high priority to search the TxPool.

Alternatively, we can modify the description of the `package work`.

**Pull request merged**